include/swift/AST/DiagnosticsSema.def

```diff
@@ -6676,6 +6676,10 @@ WARNING(executor_enqueue_deprecated_owned_job_implementation,Deprecation,
         "'Executor.enqueue(Job)' is deprecated as a protocol requirement; "
         "conform type %0 to 'Executor' by implementing 'func enqueue(ExecutorJob)' instead",
         (Type))
+WARNING(executor_enqueue_unused_implementation, none,
+        "'Executor.enqueue(ExecutorJob)' will never be used, due to the presence of "
+        "'enqueue(UnownedJob)'",
+        (Type))
```

Comment:
Noticed that you're not using the `Type` argument to this diagnostic since there's no `%0`. So you can remove it.